.github/workflows/canary.yml

```diff
@@ -0,0 +1,51 @@
+name: canary
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: build release assets
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest]
+        arch: [amd64]
+    steps:
+      - uses: actions/checkout@v2
+      - name: build release
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --all-features
+      - name: package release assets
```

Comment:
There seems to be very large overlap between canary and release. Does GitHub actions support templates like Azure DevOps does?

Reply:
it should yeah! I'll see what I can do to share code here. There are some subtle differences between the canary and release workflows, but perhaps there are ways to exclude certain jobs (or to check if it's a push event vs a tag event)

Reply:
okay, I think I managed to consolidate the two using conditionals.